### ISSUE TITLE
server: pls does not compile

### ISSUE THREAD

**Issue by camtauxe** [FractalBoy/perl-language-server#35]
I took the following steps to install the language server:

1. Clone repository and switch to `server` directory.
2. `cpanm PPI PPR Perl::Critic Perl::Tidy`
3. `perl Makefile.PL && make install`

This successfully installed the `pls` executable and associated packages into my local lib. However, attempting to run `pls` gives the following compile-time error:

```
syntax error at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 103, near "$column_number <"
Type of arg 1 to List::Util::first must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 106, near "@elements;"
syntax error at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 246, near "$column_number <="
Type of arg 1 to List::Util::any must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 447, near "->variables)
                "
Type of arg 1 to List::Util::any must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 456, near "@variables)
                    "
Type of arg 1 to List::Util::first must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 484, near "->children;"
Type of arg 1 to List::Util::first must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 1008, near "@elements;"
Type of arg 1 to List::Util::first must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 1009, near "@elements;"
Type of arg 1 to List::Util::any must be block or sub {} (not reference constructor) at PATH/TO/LOCALLIB/perl5/PLS/Parser/Document.pm line 1186, near "@installed)
            "
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS/Server/Response/Resolve.pm line 8.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS/Server/Response/Resolve.pm line 8.
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS/Server/Request/CompletionItem/Resolve.pm line 8.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS/Server/Request/CompletionItem/Resolve.pm line 8.
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS/Server/Method/CompletionItem.pm line 5.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS/Server/Method/CompletionItem.pm line 5.
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS/Server/Request/Factory.pm line 6.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS/Server/Request/Factory.pm line 6.
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS/Server.pm line 14.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS/Server.pm line 14.
Compilation failed in require at PATH/TO/LOCALLIB/perl5/PLS.pm line 5.
BEGIN failed--compilation aborted at PATH/TO/LOCALLIB/perl5/PLS.pm line 5.
Compilation failed in require at PATH/TO/LOCALBIN/pls line 12.
BEGIN failed--compilation aborted at PATH/TO/LOCALBIN/pls line 12.
```

This is with perl version 5.30.3, btw.

**Comment by FractalBoy** [comment]
I haven't tested with 5.30, so there might be some stuff I expect in core that isn't there.

I think the fix for this will be to require >= 5.32. You can install a perlbrew build just for use with your language server, even if your system perl is old.

**Comment by FractalBoy** [comment]
@camtauxe Can you try installing PLS using cpanm? I uploaded it today.

`cpanm PLS`

**Comment by stphnlyd** [comment]
It's because there is a chain comparison at this line https://github.com/FractalBoy/perl-language-server/blob/d2d0ab4f638660496241c10dfb86b2dda09c18ef/server/lib/PLS/Parser/Document.pm#L153
Chain comparison is a new feature since only perl 5.32. 

But @FractalBoy I hope we can support older versions of Perl. Afterall, older versions are still prevalent, like at our workspace we're still on Perl 5.26.

**Comment by FractalBoy** [comment]
I'd be more than happy to make it compatible with older versions - if you submit a merge request I will take a look and merge it.

**Comment by stphnlyd** [comment]
@FractalBoy I created #43

**Comment by FractalBoy** [comment]
Fixed with #43 - thank you @stphnlyd!
